packages/blast-core/lib/cudos-contract.js

```diff
@@ -10,81 +10,172 @@ const {
     }
-    const uploadTx = await this.#uploadContract()
-    const initTx = await this.#initContract(uploadTx.codeId, initMsg, this.#contractLabel, options.funds)
-    this.#contractAddress = initTx.contractAddress
+    options.signer = options.signer ?? await getDefaultLocalSigner(getNetwork(process.env.BLAST_NETWORK))
+    options.label = options.label ?? this.#label
+    const uploadTx = await this.#uploadContract(options.signer)
+    this.#codeId = uploadTx.codeId
+    this.#creator = options.signer
+    const instantiateTx = await this.#instantiateContract(
+      options.signer, uploadTx.codeId, msg, options.label, options.funds)
+    this.#label = options.label
+    this.#contractAddress = instantiateTx.contractAddress
     return {
       uploadTx: uploadTx,
-      initTx: initTx
+      initTx: instantiateTx
     }
   }
 
-  async execute(msg, signer = this.#signer) {
+  async execute(msg, signer = null) {
+    if (!this.#isDeployed()) {
+      throw new BlastError('Cannot use "execute()" on non-deployed contracts')
+    }
+    signer = signer ?? await getDefaultLocalSigner(getNetwork(process.env.BLAST_NETWORK))
     const fee = calculateFee(1_500_000, this.#gasPrice)
     return signer.execute(signer.address, this.#contractAddress, msg, fee)
   }
 
-  async query(queryMsg, signer = this.#signer) {
-    return signer.queryContractSmart(this.#contractAddress, queryMsg)
+  async query(msg, signer = null) {
+    if (!this.#isDeployed()) {
+      throw new BlastError('Cannot use "query()" on non-deployed contracts')
+    }
+    signer = signer ?? await getDefaultLocalSigner(getNetwork(process.env.BLAST_NETWORK))
+    return await signer.queryContractSmart(this.#contractAddress, msg)
   }
 
   getAddress() {
+    if (!this.#isDeployed()) {
+      return null
+    }
     return this.#contractAddress
   }
 
-  async #uploadContract() {
+  getCodeId() {
+    if (!this.#isUploaded()) {
+      return null
+    }
+    return this.#codeId
+  }
+
+  getLabel() {
+    return this.#label
+  }
+
+  getCreator() {
+    if (!this.#isUploaded()) {
+      return null
+    }
+    return this.#creator
+  }
+
+  async #uploadContract(signer) {
     // TODO: pass gasLimit as a param or read it from config
     const wasm = fs.readFileSync(this.#wasmPath)
     const uploadFee = calculateFee(1_500_000, this.#gasPrice)
-    return this.#signer.upload(
-      this.#signer.address,
+    return signer.upload(
+      signer.address,
       wasm,
       uploadFee
     )
   }
 
-  async #initContract(codeId, initMsg, label, funds) {
+  async #instantiateContract(signer, codeId, msg, label, funds) {
     const instantiateFee = calculateFee(500_000, this.#gasPrice)
-    return this.#signer.instantiate(
-      this.#signer.address,
+    return signer.instantiate(
+      signer.address,
       codeId,
-      initMsg,
+      msg,
       label,
       instantiateFee,
       { funds: funds }
     )
   }
+
+  #isUploaded() {
```

Comment:
this function is making a bit harder for me to read the code. The function is not for checking whether something is done but returning a strange value that is not expected by the name. Only one return statement in the function is not an improvement from my point of view

Reply:
fixed as discussed